cloudify_docker/tasks.py

```diff
@@ -249,58 +251,67 @@ def check_container_exited(docker_client, container):
 
 
 def move_files(source, destination, permissions=None):
-    for filename in os.listdir(source):
-        if destination == os.path.join(source, filename):
-            # moving files from parent to child case
-            # so skip
-            continue
-        shutil.move(os.path.join(source, filename),
-                    os.path.join(destination, filename))
-        if permissions:
-            os.chmod(os.path.join(destination, filename), permissions)
+    # let's handle folder vs file
+    if os.path.isdir(source):
+        for filename in os.listdir(source):
+            if destination == os.path.join(source, filename):
+                # moving files from parent to child case
+                # so skip
+                continue
+            shutil.move(os.path.join(source, filename),
+                        os.path.join(destination, filename))
+            if permissions:
+                os.chmod(os.path.join(destination, filename), permissions)
+    else:
+        shutil.move(source, destination)
 
 
 @operation
 def prepare_container_files(ctx, **kwargs):
 
     docker_ip, docker_user, docker_key, _ = get_docker_machine_from_ctx(ctx)
-    source = \
-        ctx.node.properties.get('resource_config', {}).get('source', "")
-    destination = \
-        ctx.node.properties.get('resource_config', {}).get('destination', "")
-    extra_files = \
-        ctx.node.properties.get('resource_config', {}).get('extra_files', {})
-    ansible_sources = \
-        ctx.node.properties.get(
-            'resource_config', {}).get('ansible_sources', {})
-    terraform_sources = \
-        ctx.node.properties.get(
-            'resource_config', {}).get('terraform_sources', {})
+    resource_config = ctx.node.properties.get('resource_config', {})
+    source = resource_config.get('source', "")
+    destination = resource_config.get('destination', "")
+    extra_files = resource_config.get('extra_files', {})
+    ansible_sources = resource_config.get('ansible_sources', {})
+    terraform_sources = resource_config.get('terraform_sources', {})
```

Comment:
This formula is used repeatedly:
```
    resource_config = ctx.node.properties.get('resource_config', {})
    source = resource_config.get('source', "")
    destination = resource_config.get('destination', "")
    extra_files = resource_config.get('extra_files', {})
    ansible_sources = resource_config.get('ansible_sources', {})
    terraform_sources = resource_config.get('terraform_sources', {})
```
Let's break it out into a function, for example:
```
source, dest, extra_files, ansible_sources, tf_sources = get_from_resource_config('source', 'destination', 'extra_files', 'ansible_sources', 'terraform_sources')
```

and then there will be:
```
def get_from_resource_config(*args):
    resource_config = get_resource_config()
   result = []
   for arg in args:
       item = resource_config.get(arg)
       if item:
           result.append(item)
   return result
```